### PR TITLE
[rush-lib] react to pnpm v7 local install path breaking change

### PR DESCRIPTION
**Pull request by jeremymeng**
## Summary

In pnpm v7, the format of local installation paths and the hashing algorithm used to shorten paths have been changed.  It caused error described in https://github.com/microsoft/rushstack/issues/3652. This PR adds handling logic for pnpm v7.

## How it was tested

The code is tested using https://github.com/Azure/azure-sdk-for-js repository

### CODE REVIEW

**Comment by iclanton** [comment]
Can we just use the `dependency-path` package instead of duplicating its functionality?

**Comment by jeremymeng** [comment]
> Can we just use the `dependency-path` package instead of duplicating its functionality?

@iclanton Good point. I made the change.  As far as I know our repo doesn't have such cases that are impacted by the hasing change so I don't know how to test that.  Also how do I test my PR in our repo?  Previously I was just patching the js file under node_modules.

File: common/config/rush/browser-approved-packages.json [hunk 1]
```diff
@@ -66,6 +70,10 @@
       "name": "redux",
       "allowedCategories": [ "libraries" ]
     },
+    {
+      "name": "rfc4648",
+      "allowedCategories": [ "libraries" ]
+    },
```
Comment (iclanton) [inline common/config/rush/browser-approved-packages.json:76]:
```suggestion
```

**Comment by iclanton** [comment]
You can test this change in your repo by running `node <path-to-rush-lib-in-your-clone-of-rushstack>/lib/start <args>`, or, if you're testing something that uses the caching features, use `node <path-to-rush-in-your-clone-of-rushstack>/lib/start-dev <args>`.

File: common/config/rush/browser-approved-packages.json (hunk 1 above)
Comment (jeremymeng) [reply common/config/rush/browser-approved-packages.json]: Fixed

**Comment by jeremymeng** [comment]
> You can test this change in your repo by running `node <path-to-rush-lib-in-your-clone-of-rushstack>/lib/start <args>`, or, if you're testing something that uses the caching features, use `node <path-to-rush-in-your-clone-of-rushstack>/lib/start-dev <args>`.

Thanks a lot!  Tested with our repo on Linux/Windows.

**Review by iclanton — APPROVED** [review]

**Pull request merged**